src/Parallel/Actions/TerminatePhase.hpp

```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
```

Comment:
There's also a `GotoAction.hpp` that's just in `src/Parallel/`. I'm fine with either having the `Actions` subdir or having the actions just in `src/Parallel` (though the `Actions` is more consistent). Would you mind moving the `Goto` header into the `Actions` directory?